OutOfSchool/OutOfSchool.WebApi/Models/ModelWithAdditionalData.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections.Generic;
+
+namespace OutOfSchool.WebApi.Models
+{
+    public class ModelWithAdditionalData<TModel, TData>
+        where TModel : class
+    {
+        public string Description { get; set; } = string.Empty;
+
+        public TModel Model { get; set; } = null;
+
+        public TData AdditionalData { get; set; }
```

Comment:
possible future improvement: have additional data as dictionary, so we can store multiple named additional values if needed

Reply:
Our point was, if we need a collection as AdditionalData - we set out result like "new ModelWithAdditionalData<ApplicationDto, Dictionary<string, int>>()". Or we miss something?